src/core/bonded_interactions/bonded_interaction_data.hpp

```diff
@@ -23,349 +23,114 @@
-
+/** Interaction type for virtual bonds */
+struct VirtualBond_Parameters {
+  static constexpr int num = 1;
   double cutoff() const { return -1.; }
-};
 
-/** Parameters for three-body angular potential (harmonic). */
-struct Angle_harmonic_bond_parameters {
-  /** bending constant */
-  double bend;
-  /** equilibrium angle (default is 180 degrees) */
-  double phi0;
-
-  double cutoff() const { return -1.; }
+private:
+  friend boost::serialization::access;
+  template <typename Archive>
+  void serialize(Archive &ar, long int /* version */) {}
 };
 
-/** Parameters for three-body angular potential (cosine). */
-struct Angle_cosine_bond_parameters {
-  /** bending constant */
-  double bend;
-  /** equilibrium angle (default is 180 degrees) */
-  double phi0;
-  /** cosine of @p phi0 (internal parameter) */
-  double cos_phi0;
-  /** sine of @p phi0 (internal parameter) */
-  double sin_phi0;
-
-  double cutoff() const { return -1.; }
+/** Visitor to get the number of bound partners from the bond parameter
+ *  variant */
+class BondNumPartners : public boost::static_visitor<int> {
+public:
+  template <typename T> int operator()(T const &) const { return T::num; }
 };
 
-/** Parameters for three-body angular potential (cossquare). */
-struct Angle_cossquare_bond_parameters {
-  /** bending constant */
-  double bend;
-  /** equilibrium angle (default is 180 degrees) */
-  double phi0;
-  /** cosine of @p phi0 (internal parameter) */
-  double cos_phi0;
-
-  double cutoff() const { return -1.; }
-};
-
-/** Parameters for four-body angular potential (dihedral-angle potentials). */
-struct Dihedral_bond_parameters {
-  double mult;
-  double bend;
-  double phase;
+/** Variant in which to store the parameters of an individual bonded
+ *  interaction
+ */
+using Bonded_ia_parameters = boost::variant<
+    None_bond_parameters, Fene_bond_parameters, Harmonic_bond_parameters,
+    Quartic_bond_parameters, Bonded_coulomb_bond_parameters,
+    Bonded_coulomb_sr_bond_parameters, Angle_harmonic_bond_parameters,
+    Angle_cosine_bond_parameters, Angle_cossquare_bond_parameters,
+    Dihedral_bond_parameters, Tabulated_distance_bond_parameters,
+    Tabulated_angle_bond_parameters, Tabulated_dihedral_bond_parameters,
+    Thermalized_bond_parameters, Rigid_bond_parameters, IBM_Triel_Parameters,
+    IBM_VolCons_Parameters, IBM_Tribend_Parameters,
+    Oif_global_forces_bond_parameters, Oif_local_forces_bond_parameters,
+    VirtualBond_Parameters>;
 
-  double cutoff() const { return -1.; }
-};
+/** Field containing the parameters of the bonded ia types */
+extern std::vector<Bonded_ia_parameters> bonded_ia_params;
 
-/** Parameters for n-body tabulated potential (n=2,3,4). */
-struct Tabulated_bond_parameters {
-  TabulatedBondedInteraction type;
-  TabulatedPotential *pot;
+/** These following functions are necessary to hide boost::variant from Cython.
+ *  It can't be used there because Cython doesn't support variadic templates.
+ */
```

Comment:
It's unclear to which function this doxygen block applies. You can use `@name` and delimiters `/**@{*/` and `/**@}*/` to convey that information ([example](https://github.com/espressomd/espresso/blob/a20d99b94f9ee8c798afdc56f83d51858206beb2/src/core/thermostat.hpp#L58-L67)).